### PR TITLE
Added lobby and multiplayer

### PR DESCRIPTION
**Pull request by Xue-Shangqi**
Before any of the features will run, you need to have steam open in the background since the networking uses steam's networking structure to connect everyone. If you want to test the mutliplayer with someone, you need to send the person a build of the game and let them run the game with steam open in the background. Additionally, I've only connected the Network-Lobby scene to the Gameplay scene for testing.
- [ ] Lobby (Scene: Network-Lobby)
  - [ ] Host lobby, if a player host a lobby they will be given a lobbyID that can be shared with others to let them join
  - [ ] Join lobby, requires a valid lobbyID to be able to join, will keep player on join lobby screen until they give a valid lobbyID to join
  - [ ] Leave and Start lobby, everyone can leave the lobby but only the host can start the lobby. I did not implement a feature where if host leaves, everyone will be kicked out of the lobby as of right now.
- [ ] Multiplayer (Scene: Gameplay)
  - The structure of the networking is host-client, meaning there will be one client acts as the server and process server functions. 
  - Movement is calculated a bit differently. For instance, clients will calculate their own position and pass it to the server. This will be a problem if we care about secruity since client can maniplate their positions. For now, it works since we aren't that concerned with cheaters
  - [ ] Interaction with the two cubes works and the whole inventory system
    - Invenoty is client side while interaction/drop is server side
- **IMPORTANT**
  - If you want to make an item/object accessable in multiplayer, it needs a Network Identity component. For object that needs to be transformed, it also needs a Network Transform (Reliable) componet. Also you need to do the whole server/client side updating of variables.
  - In order to spawn prefabs, they need to be added in the spawnable prefabs in NetworkManager game object.
image

- Video showcase:

https://github.com/user-attachments/assets/2e30354d-bed6-4814-a14c-5680d1c9e62e

### CODE REVIEW

File: Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+        Vector2 lookInput = _cameraSensitivity * Time.deltaTime * _playerControls.Player.Look.ReadValue<Vector2>();
+
+        // Since the axes in which we move our input device are opposite in Unity, we must swap them to ensure correct behavior.
+        // For example, moving the mouse up and/or down corresponds to side-to-side mouse movement in Unity, so we need to adjust for this.
+        _yRotation += lookInput.x;
+        _xRotation = Mathf.Clamp(_xRotation - lookInput.y, -45f, 45f); // Prevent the player from rotating their head backwards.
+
+        // "Revert" _xRotation as the controls are inverted.
+        // Without this, moving the controller down causes the camera to look up.
+        _followTransform.rotation = Quaternion.Euler(-_xRotation, _yRotation, 0);
```
Comment (christalind03) [inline Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs:122]: Currently the camera controls are inverted on the y-axis, can we fix that? (i.e: mouse down makes the camera look up)

File: Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    /// </summary>
+    /// <param name="targetPosition">The target position to set the dropped item to.</param>
+    private void Drop(Vector3? targetPosition = null)
+    {
+        GameObject droppedItem = _playerInventory.RemoveItem();
+
+        if (_raycastHit.collider != null && droppedItem != null)
+        {
+            // Determine how far the dropped item should be from the player
+            // Vector3 dropOffset = targetPosition ?? transform.position + transform.forward * 3;
```
Comment (christalind03) [inline Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs:225]: Remove comments from old code

**Review by christalind03 — CHANGES_REQUESTED** [review]
- [ ] Remove TextMeshPro folder
- [ ] Remove old comments from Monobehavior PlayerController in the Network one
- [ ] Currently the camera is inverted on the y-axis (i.e: moving the mouse downwards makes the camera look upwards)

File: Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs (hunk 1 above)
Comment (Xue-Shangqi) [reply Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs:122]: what do you mean by invert the camera? i didnt change any of the settings in the player controls. I copied the same code from the old playercontroller one and that one also had invert camera, do you want me to change it?

Comment (christalind03) [reply Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs:101]: 🤔 Maybe I'm going insane, then. If you could change it, that'd be cool. If not, no biggie

Comment (Xue-Shangqi) [reply Assets/Scripts/Network/Movement/PlayerControllerNetwork.cs:101]: Yea I can change it real fast, no worries

**Review by christalind03 — APPROVED** [review]
lgtm 💃

**Pull request merged**